### ISSUE TITLE
Use const_eval instead of constant for more lints?

### ISSUE THREAD

**Issue by llogiq** [Manishearth/rust-clippy#461]
[ ] eq_op won't join here because we expressly only want syntactically equal expressions
[x] bit_mask uses it
[x] float_cmp uses it
[ ] approx_const doesn't even need constant because we only want to match literals
[ ] zero_div_zero might probably be able to use it
[ ] min_max might use it (but we currently bail on refs, so it's debatable whether it should) 
[ ] loops could benefit from it
[ ] identity_op shouldn't use it because it bails on refs and it's so low-payoff that a full const_eval isn't worth it

What do you folks think?

**Comment by Manishearth** [comment]
We don't have a way of bypassing const resolution in const_eval though (do we?). We want to resolve constant expressions without doing const lookup because of cfgs and tweaking.

**Comment by llogiq** [comment]
Agreed. We could easily simplify `consts.rs` to work without a Context, though. const_eval should work in the other cases.

**Comment by Manishearth** [comment]
Sounds good to me
